### DISCUSSION TITLE
Fleet package repo on fleet node?

### DISCUSSION THREAD

**Discussion by OzzyKampha** [Security-Onion-Solutions/securityonion#11250] [2.4] [answered]
### Version

2.4.100

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Distributed

### Location

airgap

### Hardware Specs

Exceeds minimum requirements

### CPU

16

### RAM

30 gb

### Storage for /

1TB

### Storage for /nsm

10TB

### Network Traffic Collection

tap

### Network Traffic Speeds

more than 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Could it be an idea to have to have the elastic packages repo on the fleet node. The agents dont need to be connected to the manager. In the docs it states that a fleet node can be used in a dmz, but if the agent need to download the binary from the manager anyways it beats the porpose

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by jaypfe** [comment]
I am curious about this as well. I am not sure I fully understand exactly what binaries its talking about.  Is it just the .exe download? Or is that the config for the elastic agents? 

I think the main question is with a fleet node deployed in the DMZ, will off-network devices work as designed, or will say for instance, does the device have to be on-network to pull agent policy updates or to register with fleet the first time because of binaries?

**Comment by defensivedepth** [comment, marked as answer]
@OzzyKampha @jaypfe Unlike all the other settings on that page (see screencap), you cannot specify multiple URLs for the Agent Binary downloads. So a policy will only try to connect to a single URL. So by default we set everything to connect to the Manager to pull updates. If you have a dedicated Fleet Node with off-network agents, you will need to edit the agent policy and point their updates to the public Elastic URL. 

There is an open issue over on the Elastic repo to track adding multiple URLs.

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/954732/5b093d56-49a7-43ad-87d7-294707ec10ef)


We will add support for pulling binary updates from the Fleet Node, that is being tracked here:

https://github.com/Security-Onion-Solutions/securityonion/issues/11346

Let me know if you have any further questions.

**Reply by jaypfe** [reply]
Thank you for answering this! 

I absolutely love the continued work you guys do day in and day out to improve the Onion. Phonemical product, and its free!
